src/api/Metadata.js

```
@@ -154,26 +176,223 @@ class Metadata extends File {
+        if (hasMetadataFeature) {
+            userAddableTemplates = isExternallyOwned
+                ? [customPropertiesTemplate]
+                : [customPropertiesTemplate].concat(enterpriseTemplates);
+        }
+        return userAddableTemplates.filter(template => template.templateKey !== METADATA_TEMPLATE_CLASSIFICATION);
+    }
+
+    /**
+     * Extracts classification for different representation in the UI.
+     *
+     * @param {string} id - Box file id
+     * @param {Array} instances - metadata instances
+     * @return {Array} metadata instances without classification
+     */
+    extractClassification(id: string, instances: Array<MetadataInstance>): Array<MetadataInstance> {
+        const classification = instances.find(instance => instance.$template === METADATA_TEMPLATE_CLASSIFICATION);
+        if (classification) {
+            instances.splice(instances.indexOf(classification), 1);
+            const cache: APICache = this.getCache();
+            const key = this.getClassificationCacheKey(id);
+            cache.set(key, classification);
+        }
+        return instances;
+    }
+
+    /**
+     * Finds template for a given metadata instance.
+     *
+     * @param {string} id - Box file id
+     * @param {Object} instance - metadata instance
+     * @param {Array} templates - metadata templates
+     * @return {Object|undefined} template for metadata instance
+     */
+    async getTemplateForInstance(
+        id: string,
+        instance: MetadataInstance,
+        templates: Array<MetadataEditorTemplate>,
+    ): Promise<?MetadataEditorTemplate> {
+        const instanceId = instance.$id;
+        const templateKey = instance.$template;
+        const scope = instance.$scope;
+        let template = templates.find(t => t.templateKey === templateKey && t.scope === scope);
+
+        // Enterprise scopes are always enterprise_XXXXX
+        if (scope.startsWith(METADATA_SCOPE_ENTERPRISE) && !template) {
```

minor: consider swapping the order of the condition so `!template` could fail faster